include/zephyr/drivers/video-controls.h

```diff
@@ -58,15 +70,16 @@ extern "C" {
  * @name Camera class control IDs
  * @{
  */
-#define VIDEO_CID_CAMERA_EXPOSURE	(VIDEO_CTRL_CLASS_CAMERA + 0)
-#define VIDEO_CID_CAMERA_GAIN		(VIDEO_CTRL_CLASS_CAMERA + 1)
-#define VIDEO_CID_CAMERA_ZOOM		(VIDEO_CTRL_CLASS_CAMERA + 2)
-#define VIDEO_CID_CAMERA_BRIGHTNESS	(VIDEO_CTRL_CLASS_CAMERA + 3)
-#define VIDEO_CID_CAMERA_SATURATION	(VIDEO_CTRL_CLASS_CAMERA + 4)
-#define VIDEO_CID_CAMERA_WHITE_BAL	(VIDEO_CTRL_CLASS_CAMERA + 5)
-#define VIDEO_CID_CAMERA_CONTRAST	(VIDEO_CTRL_CLASS_CAMERA + 6)
-#define VIDEO_CID_CAMERA_COLORBAR	(VIDEO_CTRL_CLASS_CAMERA + 7)
-#define VIDEO_CID_CAMERA_QUALITY	(VIDEO_CTRL_CLASS_CAMERA + 8)
+#define VIDEO_CID_CAMERA_EXPOSURE     (VIDEO_CTRL_CLASS_CAMERA + 0)
+#define VIDEO_CID_CAMERA_GAIN         (VIDEO_CTRL_CLASS_CAMERA + 1)
+#define VIDEO_CID_CAMERA_ZOOM         (VIDEO_CTRL_CLASS_CAMERA + 2)
+#define VIDEO_CID_CAMERA_BRIGHTNESS   (VIDEO_CTRL_CLASS_CAMERA + 3)
+#define VIDEO_CID_CAMERA_SATURATION   (VIDEO_CTRL_CLASS_CAMERA + 4)
+#define VIDEO_CID_CAMERA_WHITE_BAL    (VIDEO_CTRL_CLASS_CAMERA + 5)
+#define VIDEO_CID_CAMERA_CONTRAST     (VIDEO_CTRL_CLASS_CAMERA + 6)
+#define VIDEO_CID_CAMERA_TEST_PATTERN (VIDEO_CTRL_CLASS_CAMERA + 7)
```

Comment:
I hope other hardware than camera devices will have a test pattern generator (i.e. VPUs, compression engines).
Would it be something to add to the "generic" control class?

Reply:
Yes, I think it should not belong to CAMERA class too. However, it's a bit blur how to categorize the classes and it's a bit different with what are done in Linux. For example, should we create an IMAGE_PROC class and move CID_TEST_PATTERN, CID_PIXEL_RATE into this class as in Linux ... The same for CID_BRIGHTNESS, CONTRAST, HUE, SATURATION, CID_GAIN etc. which are in BASE class in Linux (equivalent to GENERIC class in Zephyr) ?

Reply:
Various protocols will also introduce their own control categories (i.e. in USB UVC: Processing Unit, Encoding Unit, Camera Terminal, Extension Unit, probably some others for GigE-Vision, and others in V4L2) and be mapped to Zephyr controls.

Good topic for another PR/issue and does not look blocking...
Thank you for your patience!

Reply:
I tried this based on your feedback. Is that helping?
- https://github.com/zephyrproject-rtos/zephyr/pull/78802

Reply:
This sounds good, I intended to do so too (and also some changes in the control API) but never had time to complete this. So, perhaps I will take this and keep only the control API change part.